src/opensynth/models/faraday/model.py

```diff
@@ -329,6 +345,90 @@ def __init__(
+    ) -> dict[str, dict[str, int]]:
+        """
+        Get the max and min values of numerically encoded features
+        Args:
+            features (dict[str, torch.tensor]): Dictionary of
+            feature tensors
+        Returns:
+            dict[str, dict[str, int]]: A dictionary of
+            min and max values for each feature label
+        """
+        feature_range: dict[str, dict[str, int]] = {}
+        for feature in features:
+            feature_range[feature] = {
+                "min": features[feature].min().item(),
+                "max": features[feature].max().item(),
+            }
+        return feature_range
+
+    @staticmethod
+    def create_mask(
+        gmm_labels: dict[str, torch.tensor],
+        range_dict: dict[str, dict[str, int]],
+    ) -> npt.NDArray[np.bool_]:
+        """
+        Create a filter mask to make sure that GMM sampled labels are within
+        the range of accepted values. There is no guarantee that GMM will not
+        create out-of-distribution values. For example if `month of year`
+        ranges from 0-11, there is no guarantee that GMM will not
+        result in `month of year` = 100.
+        Args:
+            gmm_labels (dict[str, torch.tensor]): Dictionary of features
+            obtained from GMM model
+            range_dict (dict[str, dict[str, int]]): Dictionary of
+            ranges of each label
+        Returns:
+            list[bool]: Mask to filter out-of-distribution values
+        """
+        label_mask: Optional[npt.NDArray[np.bool_]] = None  # Initialize mask
+
+        for feature, bounds in range_dict.items():
+
+            # Fetch the feature range
+            min_value = bounds.get("min")
+            max_value = bounds.get("max")
+            if min_value is None or max_value is None:
+                raise ValueError(f"Feature {feature} have missing range")
+
+            # Dynamically fetch the respective labels
+            gmm_value = gmm_labels.get(feature)
+            if gmm_value is None:
+                raise ValueError(f"Feature {feature} not found in GMM labels")
+
+            # Create the mask for this label
+            current_mask = np.logical_and(
+                gmm_value.detach().numpy() >= min_value,
+                gmm_value.detach().numpy() <= max_value,
+            )
+
+            # Combine the masks with `&` (AND operation)
+            label_mask = (
+                current_mask
+                if label_mask is None
+                else (label_mask & current_mask)
+            )
+
+        return label_mask
+
+    @staticmethod
+    def get_index(feature_list: list[str], current_index: int) -> int:
+        """
+        Get the index of each label
+
+        Args:
+            feature_list (list[str]): List of features
+            current_index (int): Current iter index
+
+        Returns:
+            int: Returns colum index
+        """
+        return -(len(feature_list) - current_index)
+
```

Comment:
I feel that it would be cleaner to have these static modules (`get_feature_range`, `create_mask`, `get_index`) taken out of FaradayModel into their own class, given that they have a specific purpose of filtering out invalid samples.

Reply:
But these functions are also specific to `FaradayModel` only though, and doesn't make sense as a standalone. Taking it out makes this out of context.

We can always take this out into a separate utils.py if other classes of model require this.